.github/workflows/build.yaml

```diff
@@ -0,0 +1,165 @@
+      - name: Compress build
+        working-directory: ${{github.workspace}}
+        run: tar --xz -cf /tmp/archive.tar.xz --exclude '*.o' --exclude '*.pyc' --exclude '.git' --exclude='qpid-dispatch/build/console' qpid-dispatch install qpid-proton/build/python/pkgs
+      - name: Upload archive
+        uses: actions/upload-artifact@v2
+        with:
+          name: qpid_dispatch_wrk_${{matrix.os}}_${{matrix.buildType}}
+          path: /tmp/archive.tar.xz
+  test:
+    name: "Test (${{matrix.os}}, shard ${{matrix.shard}} of ${{matrix.shards}}"
+    runs-on: ${{ matrix.os }}
+    needs: [build]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+        buildType: [RelWithDebInfo]
+        shard: [1, 2, 3, 4]
+        shards: [4]
+    env:
+      BuildType: ${{matrix.buildType}}
+      ProtonBuildDir: ${{github.workspace}}/qpid-proton/build
+      DispatchBuildDir: ${{github.workspace}}/qpid-dispatch/build
+      InstallPrefix: ${{github.workspace}}/install
+      PKG_CONFIG_PATH: ${{matrix.pkg_config_path}}
+      LD_LIBRARY_PATH: ${{github.workspace}}/install/lib
+    steps:
+      - name: Download Build
+        uses: actions/download-artifact@v2
+        with:
+          name: qpid_dispatch_wrk_${{matrix.os}}_${{matrix.buildType}}
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+          architecture: x64
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox
+      - name: Install Linux dependencies
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo apt install -y bubblewrap swig ninja-build libsasl2-2 sasl2-bin libjsoncpp1 libwebsockets15 xz-utils
+      - name: Install MacOS dependencies
+        if: ${{ runner.os == 'macOS' }}
+        run: |
+          brew install cmake ninja libuv swig pkgconfig jsoncpp openssl libwebsockets
+      - name: Unpack archive
+        run: tar --xz -xf archive.tar.xz
+      - name: install qpid-proton python wheel
+        run: python -m pip install ${ProtonBuildDir}/python/pkgs/python_qpid_proton*.whl
+        shell: bash
+      - name: CTest
+        working-directory: ${{env.DispatchBuildDir}}
+        run: |
+          ulimit -c unlimited
+          ctest -C ${BuildType} -V -T Test --no-compress-output -I ${{matrix.shard}},,${{matrix.shards}} -j2
+        shell: bash
+      - name: Upload test results
+        uses: actions/upload-artifact@v2
+        if: ${{ ! cancelled() }}
+        with:
+          name: Test_Results_${{matrix.os}}_${{matrix.buildType}}_${{matrix.shard}}
+          path: ${{env.DispatchBuildDir}}/Testing/**/*.xml
+      - name: Upload core files
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: cores_${{matrix.os}}_${{matrix.buildType}}_${{matrix.shard}}
+          path: |
+            **/core
+      - name: Environment (Linux)
```

Comment:
Would this be as well done at the start / before testing, rather than the end?

Reply:
Proton runs this at the end. It probably doesn't matter that much if it's at the end or the beginning, given that the tests will be running really fast ;)

Reply:
Same comment would apply to proton if I'd thought of it at the time :) Just seems more usual to care about the env at the start rather than end, e.g all the stuff about the node etc is usually at the start.

Reply:
Moved to the beginning.